### PR TITLE
Visuals for the subscription (mobile and desktop)

### PR DESCRIPTION
**Pull request by delapuente**

### CODE REVIEW

File: wp-web-push/lib/js/wp-web-push.js [hunk 1]
```diff
@@ -215,60 +241,96 @@ if (navigator.serviceWorker) {
...
...p)|hs\-c|ht(c(\-| |_|a|g|p|s|t)|tp)|hu(aw|tc)|i\-(20|go|ma)|i230|iac( |\-|\/)|ibro|idea|ig01|ikom|im1k|inno|ipaq|iris|ja(t|v)a|jbro|jemu|jigs|kddi|keji|kgt( |\/)|klon|kpt |kwc\-|kyo(c|k)|le(no|xi)|lg( g|\/(k|l|u)|50|54|\-[a-w])|libw|lynx|m1\-w|m3ga|m50\/|ma(te|ui|xo)|mc(01|21|ca)|m\-cr|me(rc|ri)|mi(o8|oa|ts)|mmef|mo(01|02|bi|de|do|t(\-| |o|v)|zz)|mt(50|p1|v )|mwbp|mywa|n10[0-2]|n20[2-3]|n30(0|2)|n50(0|2|5)|n7(0(0|1)|10)|ne((c|m)\-|on|tf|wf|wg|wt)|nok(6|i)|nzph|o2im|op(ti|wv)|oran|owg1|p800|pan(a|d|t)|pdxg|pg(13|\-([1-8]|c))|phil|pire|pl(ay|uc)|pn\-2|po(ck|rt|se)|prox|psio|pt\-g|qa\-a|qc(07|12|21|32|60|\-[2-7]|i\-)|qtek|r380|r600|raks|rim9|ro(ve|zo)|s55\/|sa(ge|ma|mm|ms|ny|va)|sc(01|h\-|oo|p\-)|sdk\/|se(c(\-|0|1)|47|mc|nd|ri)|sgh\-|shar|sie(\-|m)|sk\-0|sl(45|id)|sm(al|ar|b3|it|t5)|so(ft|ny)|sp(01|h\-|v\-|v )|sy(01|mb)|t2(18|50)|t6(00|10|18)|ta(gt|lk)|tcl\-|tdg\-|tel(i|m)|tim\-|t\-mo|to(pl|sh)|ts(70|m\-|m3|m5)|tx\-9|up(\.b|g1|si)|utst|v400|v750|veri|vi(rg|te)|vk(40|5[0-3]|\-v)|vm40|voda|vulc|vx(52|53|60|61|70|80|81|83|85|98)|w3c(\-| )|webc|whit|wi(g |nc|nw)|wmlb|wonu|x700|yas\-|your|zeto|zte\-/i.test(a.substr(0,4)))check = true;})(navigator.userAgent||navigator.vendor||window.opera);
```
Comment (marco-c) [inline wp-web-push/lib/js/wp-web-push.js:247]: We should do the same for tablets, the regex is in the same StackOverflow answer.

File: wp-web-push/lib/js/wp-web-push.js [hunk 2]
```diff
@@ -215,60 +241,96 @@ if (navigator.serviceWorker) {
...
+      document.querySelector('#webpush-subscription .subscribe').onclick = function() {
+        localforage.setItem('button_interacted', true);
+        setNotificationsIndicator(false);
+
+        notificationsEnabled()
+          .then(function(enabled) {
+            if (enabled && Notification.permission === 'granted') {
+              showUnsubscribe();
+            } else {
+              showSubscribe();
```
Comment (marco-c) [inline wp-web-push/lib/js/wp-web-push.js:321]: Also need to call enableNotifications(true) here, otherwise we're not subscribing the user.

Comment (marco-c) [inline wp-web-push/lib/js/wp-web-push.js:321]: The onclick for this button and the subscription icon in the desktop case are really similar, we should put the common code in a function (here or in a follow-up).

File: wp-web-push/lib/js/wp-web-push.js [hunk 3]
```diff
@@ -38,16 +38,42 @@ if (navigator.serviceWorker) {
     }, 200);
   }
 
+  function showSubscribe() {
+    document.querySelector('#webpush-subscription .message').innerHTML = '<p>' + ServiceWorker.subscription_prompt + '</p><p><img src="' + ServiceWorker.notification_preview + '" alt="" /></p>';
+    var actionButton = document.querySelector('#webpush-subscription .actions .default');
+    actionButton.innerHTML = ServiceWorker.subscription_button_text;
```
Comment (marco-c) [inline wp-web-push/lib/js/wp-web-push.js]: Nit: we could use `textContent` in some cases.

File: wp-web-push/lib/js/wp-web-push.js (hunk 1 above)
Comment (delapuente) [inline wp-web-push/lib/js/wp-web-push.js:247]: It's that, I used the name `isMobile` but I'm using the other one for tablet detection.

File: wp-web-push/lib/js/wp-web-push.js (hunk 2 above)
Comment (delapuente) [inline wp-web-push/lib/js/wp-web-push.js:321]: You don't want to subscribe here, you want to show the dialog to give the user the chance of cancelling the subscription. If not, why do you use a dialog at all?

File: wp-web-push/lib/js/wp-web-push.js (hunk 1 above)
Comment (marco-c) [inline wp-web-push/lib/js/wp-web-push.js:247]: :+1:

File: wp-web-push/lib/js/wp-web-push.js (hunk 2 above)
Comment (marco-c) [inline wp-web-push/lib/js/wp-web-push.js:321]: The subscription isn't working for me in the mobile case, so I thought this was the button, sorry.

File: wp-web-push/lib/js/wp-web-push.js [hunk 4]
```diff
@@ -38,16 +38,42 @@ if (navigator.serviceWorker) {
...
+    actionButton.onclick = function () {
+      enableNotifications()
```
Comment (marco-c) [inline wp-web-push/lib/js/wp-web-push.js]: Use `enableNotifications(true)` here (`true` means that we ignore the interval between prompts, which we should do if the user explicitly clicks on the button).

File: wp-web-push/lib/js/wp-web-push.js [hunk 5]
```diff
@@ -38,16 +38,42 @@ if (navigator.serviceWorker) {
...
+      .then(dismissDialog);
+    };
+    document.querySelector('#webpush-subscription .dialog').classList.add('shown');
+  }
+
+  function showUnsubscribe() {
+    document.querySelector('#webpush-subscription .message').innerHTML = '<p>' + ServiceWorker.unsubscription_prompt + '</p>';
+    var actionButton = document.querySelector('#webpush-subscription .actions .default');
+    actionButton.innerHTML = ServiceWorker.unsubscription_button_text;
```
Comment (marco-c) [inline wp-web-push/lib/js/wp-web-push.js]: Nit: textContent

File: wp-web-push/lib/js/wp-web-push.js [hunk 6]
```diff
@@ -215,60 +241,96 @@ if (navigator.serviceWorker) {
...
+                  setSubscriptionTip(ServiceWorker.unsubscription_hint);
+                });
+            }
+          });
+      };
+    }
+    else {
+      var closeButton = document.querySelector('#webpush-subscription .actions .dismiss');
+
+      closeButton.innerHTML = ServiceWorker.close_button_text;
```
Comment (marco-c) [inline wp-web-push/lib/js/wp-web-push.js]: Nit: textContent

**Pull request merged**